docs/tutorials/prometheus.ipynb

```diff
@@ -0,0 +1,471 @@
+        "\n",
+        "!dig @127.0.0.1 -p 9053 demo1.example.org\n",
+        "!dig @127.0.0.1 -p 9053 demo2.example.org"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Afb01zkACsuW",
+        "colab_type": "text"
+      },
+      "source": [
+        "The next step is to setup Prometheus server and use Prometheus to scrape CoreDNS metrics that are exposed on port `9153` from above. The `prometheus.yml` file for configuration is also available for [download](https://github.com/tensorflow/io/blob/master/docs/tutorials/prometheus/prometheus.yml):\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "3HHfVdv9D3Jv",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "!curl -s -OL https://github.com/prometheus/prometheus/releases/download/v2.15.2/prometheus-2.15.2.linux-amd64.tar.gz\n",
+        "!tar -xzf prometheus-2.15.2.linux-amd64.tar.gz --strip-components=1\n",
+        "\n",
+        "!curl -s -OL https://raw.githubusercontent.com/tensorflow/io/master/docs/tutorials/prometheus/prometheus.yml\n",
+        "\n",
+        "# Run `./prometheus` as a background process.\n",
+        "# IPython doesn't recognize `&` in inline bash cells.\n",
+        "get_ipython().system_raw('./prometheus &')"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "f61fK3bXQH4N",
+        "colab_type": "text"
+      },
+      "source": [
+        "Now a CoreDNS server whose metrics are scraped by a Prometheus server and ready to be consumed by TensorFlow."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "acEST3amdyDI",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Create Dataset for CoreDNS metrics and use it in TensorFlow\n",
+        "\n",
+        "Create a Dataset for CoreDNS metrics that is available from PostgreSQL server, could be done with `tfio.experimental.IODataset.from_prometheus`. At the minimium two arguments are needed. `query` is passed to Prometheus server to select the metrics and `length` is the period we want to load into Dataset.\n",
+        "\n",
+        "You can start with \"coredns_dns_request_count_total\" and \"5\" (secs) to create the Dataset below. Since earlier in the tutorial two DNS queries were sent, it is expected that the metrics for \"coredns_dns_request_count_total\" will be \"2.0\" at the end of the time series:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "h21RdP7meGzP",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "from datetime import datetime\n",
+        "import tensorflow_io as tfio\n",
+        "\n",
+        "dataset = tfio.experimental.IODataset.from_prometheus(\n",
```

Comment:
Getting an error here when run on Colab. Is this expected?

@MarkDaoust Since this notebook is spinning up servers and such, will this work when imported into tensorflow.org? Do we need to save the outputs?

Trace:
```
WARNING:tensorflow:AutoGraph could not transform <function io_prometheus_readable_read at 0x7f5de5f60e18> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: Unreachable
WARNING: AutoGraph could not transform <function io_prometheus_readable_read at 0x7f5de5f60e18> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: Unreachable
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-6-5ea8e8be172d> in <module>()
      3 
      4 dataset = tfio.experimental.IODataset.from_prometheus(
----> 5     "coredns_dns_request_count_total", 5)
      6 
      7 print("Dataset Spec:\n{}\n".format(dataset.element_spec))

12 frames
/tensorflow-2.1.0/python3.6/tensorflow_core/python/autograph/impl/api.py in wrapper(*args, **kwargs)
    235       except Exception as e:  # pylint:disable=broad-except
    236         if hasattr(e, 'ag_error_metadata'):
--> 237           raise e.ag_error_metadata.to_exception(e)
    238         else:
    239           raise

AssertionError: in converted code:

    /usr/local/lib/python3.6/dist-packages/tensorflow_io/core/python/ops/prometheus_dataset_ops.py:82 f  *
        timestamp, value = golang_ops.io_prometheus_readable_read(
    <string>:254 io_prometheus_readable_read
        
    /tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/op_def_library.py:602 _apply_op_helper
        assert False, "Unreachable"

    AssertionError: Unreachable
```

Reply:
> Since this notebook is spinning up servers and such, will this work when imported into tensorflow.org?

I think this part shouldn't be a problem.